### PR TITLE
Prevent individual errors from validating a list's items from being lost.

### LINKED ISSUE #150: list validation swallows more useful/explicit error messages
Take this validator and schema:

```python
def is_even(value):
    if value % 2:
        raise Invalid('%i is not even' % value)
    return value

schema = Schema(dict(even_number=All(int, is_even)))
schema(dict(even_number=3))
```

We get a useful error message:

```
voluptuous.MultipleInvalid: 3 is not even for dictionary value @ data['even_number']
```

Sadly, as soon as we want a list of these, we lose the useful error message:

```python
schema = Schema(dict(even_numbers=[All(int, is_even)]))
schema(dict(even_numbers=[3]))
````

Instead, we get a much less useful message:

```
voluptuous.MultipleInvalid: invalid list value @ data['even_numbers'][0]
```

Sadly, nowhere in that `MultipleInvalid` exception is the original error stored.

### PR DESCRIPTION
**Pull request by cjw296**
This fixes #150.

### CODE REVIEW

File: voluptuous.py [hunk 1]
```diff
@@ -699,6 +699,8 @@ def validate_sequence(path, data):
                         invalid = e
                 else:
                     if len(invalid.path) <= len(index_path):
+                        if invalid is not None:
```
Comment (alecthomas) [inline voluptuous.py]: This can never be `None` due to the previous `if len(invalid.path) ...` but it seems like it should be checked somewhere.

**Comment by cjw296** [comment]
f1c9ef3 is the commit I referred to in the email I sent, let me know your thoughts on this one, I'm not 100%, but as you can see, it only required minimal changes to the test expectations.

**Comment by alecthomas** [comment]
Looks good to me. If it solves your problem and doesn't introduce any others, I'm fine with it :)

**Comment by cjw296** [comment]
@alecthomas - great, thanks! Could I nudge you for a release?

(Also, in passing, it doesn't look like 0.8.9 has been tagged in the git repo...)

**Comment by alecthomas** [comment]
[Done](https://pypi.python.org/pypi/voluptuous/0.8.10).

Also: 0.8.9 was tagged locally, but not pushed.

**Comment by cjw296** [comment]
Fantastic, thanks!

**Pull request merged**